tidbcloud/api.go

```diff
@@ -0,0 +1,244 @@
+	if err != nil {
+		return nil, err
+	}
+
+	return &result, nil
+}
+
+// CreateDedicatedCluster create a dedicated cluster in the given project
+func (c *TiDBCloudClient) CreateDedicatedCluster(projectID string, spec *Specification) (*CreateClusterResp, error) {
+	var (
+		url    = fmt.Sprintf("%s/api/v1beta/projects/%s/clusters", host, projectID)
+		result CreateClusterResp
+	)
+
+	// We have check the boundary in main function
+	tidbSpec := spec.Tidb[0]
+	tikvSpec := spec.Tikv[0]
+
+	payload := CreateClusterReq{
+		Name:          "tidbcloud-sample-1", // NOTE change to your cluster name
+		ClusterType:   spec.ClusterType,
+		CloudProvider: spec.CloudProvider,
+		Region:        spec.Region,
+		Config: ClusterConfig{
+			RootPassword: "your secret password", // NOTE change to your cluster password, we generate a random password here
+			Port:         4000,
+			Components: Components{
+				TiDB: ComponentTiDB{
+					NodeSize:     tidbSpec.NodeSize,
+					NodeQuantity: tidbSpec.NodeQuantityRange.Min,
+				},
+				TiKV: ComponentTiKV{
+					NodeSize:       tikvSpec.NodeSize,
+					StorageSizeGib: tikvSpec.StorageSizeGibRange.Min,
+					NodeQuantity:   tikvSpec.NodeQuantityRange.Min,
+				},
+			},
+			IPAccessList: []IPAccess{
+				{
+					CIDR:        "0.0.0.0/0",
+					Description: "Allow Access from Anywhere.",
+				},
+			},
+		},
+	}
+
+	_, err := doPOST(url, payload, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	return &result, nil
+}
+
+// CreateCluster create a cluster in the given project
+func (c *TiDBCloudClient) CreateCluster(projectID string, clusterReq *CreateClusterReq) (*CreateClusterResp, error) {
+	var (
+		url    = fmt.Sprintf("%s/api/v1beta/projects/%s/clusters", host, projectID)
+		result CreateClusterResp
+	)
+
+	_, err := doPOST(url, clusterReq, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	return &result, nil
+}
+
+// GetClusterById return detail status of given cluster
+func (c *TiDBCloudClient) GetClusterById(projectID string, clusterID string) (*GetClusterResp, error) {
+	var (
+		url    = fmt.Sprintf("%s/api/v1beta/projects/%s/clusters/%s", host, projectID, clusterID)
+		result GetClusterResp
+	)
+
+	_, err := doGET(url, nil, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	return &result, nil
+}
+
+// DeleteClusterById delete a cluster by the given ID
+func (c *TiDBCloudClient) DeleteClusterById(projectID, clusterID string) error {
+	url := fmt.Sprintf("%s/api/v1beta/projects/%s/clusters/%s", host, projectID, clusterID)
+	_, err := doDELETE(url, nil, nil)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// UpdateClusterById can only scale out and set pause now
+func (c *TiDBCloudClient) UpdateClusterById(projectID, clusterID string, updateClusterReq UpdateClusterReq) error {
+	url := fmt.Sprintf("%s/api/v1beta/projects/%s/clusters/%s", host, projectID, clusterID)
+	resp, err := doPATCH(url, updateClusterReq, nil)
+	if err != nil {
+		return err
+	}
+	print(resp)
```

Comment:
why some where is log.Printf(), here is print

Reply:
I forger to delete it